api/api.go

```diff
@@ -549,6 +555,7 @@ func (r *request) setQueryOptions(q *QueryOptions) {
 	for k, v := range q.Params {
 		r.params.Set(k, v)
 	}
+	r.ctx = q.Context()
```

Comment:
This is the only caller and it's for a concern that's internal to this file... I think inlining the `Context()` method here is the way to go.

Reply:
Do you want me to delete the `Context` method entirely or leave it in the API so that API consumers can use it?

I added it because it was available in Consul https://github.com/hashicorp/consul/blob/master/api/api.go#L171
and used it here because I figured it was good practice to use consumer-facing functions internally when possible.

That said happy to change just let me know

Reply:
Nope, let's keep it! See my remarks in https://github.com/hashicorp/nomad/pull/8836/files#r485614197.